src/wireviz/Harness.py

```diff
@@ -394,24 +456,77 @@ def bom(self):
             name = f'Wire{wire_type}{gauge_name}{gauge_color}'
             item = {'item': name, 'qty': round(total_length, 3), 'unit': 'm', 'designators': designators,
                     'manufacturer': shared['manufacturer'], 'mpn': shared['mpn'], 'pn': shared['pn']}
-            bom_cables.append(item)
-            bom_cables = sorted(bom_cables, key=lambda k: k['item'])  # sort list of dicts by their values (https://stackoverflow.com/a/73050)
-        bom.extend(bom_cables)
+            bom_items.append(item)
+
+        for cable in self.cables.values():
+            if cable.additional_components:
+                for part in cable.additional_components:
+                    qty = part.get('qty', 1)
+                    if 'qty_multiplier' in part:
+                        if part['qty_multiplier'] == 'wirecount':
+                            qty *= cable.wirecount
+                        elif part['qty_multiplier'] == 'terminations':
+                            qty *= len(cable.connections)
+                        elif part['qty_multiplier'] == 'length':
+                            qty *= cable.length
+                        elif part['qty_multiplier'] == 'total_length':
+                            qty *= cable.length * cable.wirecount
+                        else:
+                            raise ValueError('invalid qty parameter {}'.format(part["qty_multiplier"]))
+                    bom_items.append(
+                        {
+                            'item': extra_component_long_name(part["type"], part.get("subtype", None)),
+                            'qty': qty,
+                            'unit': part.get('unit', None),
+                            'manufacturer': part.get('manufacturer', None),
+                            'mpn': part.get('mpn', None),
+                            'pn': part.get('pn', None),
+                            'designators': cable.name
+                        }
+                    )
 
         for item in self.additional_bom_items:
             name = item['description'] if item.get('description', None) else ''
-            if isinstance(item.get('designators', None), List):
-                item['designators'].sort()  # sort designators if a list is provided
             item = {'item': name, 'qty': item.get('qty', None), 'unit': item.get('unit', None), 'designators': item.get('designators', None),
                     'manufacturer': item.get('manufacturer', None), 'mpn': item.get('mpn', None), 'pn': item.get('pn', None)}
```

Comment:
I suggest removing all 7 `None` arguments in the 4 lines above. The second parameter of `.get()` is optional with default value`=None`. Simpler code is normally easier to read.